lib/tbd/geo.rb

```diff
@@ -722,15 +722,10 @@ def resetKIVA(model = nil, boundary = "Foundation")
       kva = true unless surface.surfacePropertyExposedFoundationPerimeter.empty?
       surface.resetAdjacentFoundation
       surface.resetSurfacePropertyExposedFoundationPerimeter
-      next unless surface.isGroundSurface
       next     if surface.outsideBoundaryCondition.capitalize == boundary
+      next unless surface.outsideBoundaryCondition.capitalize == "Foundation"
```

Comment:
Boils down to either "Ground"-facing or "Foundation"-facing surfaces. If purging KIVA objects WITHOUT starting over, then "Foundation"-facing surfaces have their outside boundary condition reset to "Ground" - much safer. If instead purging KIVA objects THEN starting over, both "Ground"-facing and "Foundation"-facing surfaces are left as is. It's valid to have a model with one space having "Ground"-facing surfaces, with another having "Foundation"-facing surfaces. No longer any need to (re)set constructions.